components/script/dom/eventtarget.rs

```diff
@@ -315,11 +315,35 @@ impl EventTarget {
     pub fn dispatch_event_with_target(&self,
                                       target: &EventTarget,
                                       event: &Event) -> EventStatus {
-        event.dispatch(self, Some(target))
```

Comment:
Assert here, and let's match on `target.global().downcast::<Window>()` instead of trying Element (keeping in mind that None is valid).